backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -10,24 +10,63 @@
 
 import gov.cdc.usds.simplereport.api.model.errors.IllegalGraphqlArgumentException;
 import gov.cdc.usds.simplereport.config.simplereport.AdminEmailList;
+import gov.cdc.usds.simplereport.db.model.auxiliary.PersonName;
 import gov.cdc.usds.simplereport.db.model.ApiUser;
 import gov.cdc.usds.simplereport.db.repository.ApiUserRepository;
 import gov.cdc.usds.simplereport.service.model.IdentityAttributes;
 import gov.cdc.usds.simplereport.service.model.IdentitySupplier;
+import gov.cdc.usds.simplereport.service.OktaService;
 
 @Service
 @Transactional
 public class ApiUserService {
+
     private AdminEmailList _admins;
     private ApiUserRepository _apiUserRepo;
     private IdentitySupplier _supplier;
+    private OktaService _oktaService;
 
     private static final Logger LOG = LoggerFactory.getLogger(ApiUserService.class);
-
-    public ApiUserService(ApiUserRepository apiUserRepo, IdentitySupplier supplier, AdminEmailList admins) {
+    
+    public ApiUserService(
+        ApiUserRepository apiUserRepo,
+        IdentitySupplier supplier,
+        AdminEmailList admins,
+        OktaService oktaService
+    ) {
         _apiUserRepo = apiUserRepo;
         _supplier = supplier;
         _admins = admins;
+        _oktaService = oktaService;
+    }
+
+    @Transactional
+    public ApiUser createUser(String username, String firstName, String middleName, String lastName, String suffix, String organizationExternalId) {
+        isAdminUser();
+        IdentityAttributes userIdentity = new IdentityAttributes(username, firstName, middleName, lastName, suffix);
+        ApiUser user = _apiUserRepo.save(new ApiUser(username, userIdentity));
```

Comment:
"the ApiUser object should get created when the user first logs in"

if i do that, how should that get exposed? i'm assuming i should take the "createUser" mutation out of the graphql layer, but then what query/mutation would trigger ApiUser::createUser()?

Reply:
It currently happens in ApiUserService.getCurrentUser (which is actually not called every time somebody visits, so last_seen is probably a little squirrelly—we should probably fix that.)

Reply:
> I actually don't think we want to do this: the ApiUser object should get created when the user first logs in, and otherwise should not IMO be created—if we do want to create it, we need to create it with a flag value (or null if we allow that) in the `last_seen` field, since we have in fact never seen them.

clarification question -- when you say "we don't want to do this", do you mean "expose a createUser mutation in the graphql layer at all" ?

Reply:
This is a good feature. We should be able to create users in our system and in okta before they login successfully. `last_seen=null` seems like a great default and initial state for a user who has not logged in